src/classes/RD2_ValidationService.cls

```diff
@@ -280,6 +281,22 @@ public with sharing class RD2_ValidationService {
             && oldAcct.npe01__SYSTEMIsIndividual__c == false;
     }
 
+    /***
+    * @description Return true when account is changed from null to contact's account and contact is not changed.
+    * @param new recurring donation record
+    * @param old recurring donation record
+    * @return Boolean True when account is updated to contact's account from null value
+    */
+    private Boolean isAccountUpdatedToContactAccount(npe03__Recurring_Donation__c rd, npe03__Recurring_Donation__c oldRd) {
+        Map<Id,Id> rdWithAccIdFromContact = getAccountIdFromContactInRecurringDonations();
```

Comment:
The SOQL was being executed for every RD, so using the `contactById` as the property can avoid limit errors.